### PR TITLE
US914108: Version Currency: JUnit 5 migration

### PR DESCRIPTION
**Pull request by raghavendran-gopalakrishnan**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=914108

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -203,9 +203,9 @@
                 <version>3.1.0</version>
             </dependency>
             <dependency>
-                <groupId>junit</groupId>
-                <artifactId>junit</artifactId>
-                <version>4.13.2</version>
+                <groupId>org.junit.jupiter</groupId>
```
Comment (dermot-hardy) [inline pom.xml]: Dependency management should be in alphabetical order.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
The main branch build shows 42 tests but your build only shows 41:

- https://sou-jenkins2.swinfra.net/job/CAFapi/job/CAFapi~caf-common~develop~CI/88/testReport/
- https://sou-jenkins2.swinfra.net/job/CAFapi/job/CAFapi~caf-common~914108~CI/2/testReport/

Why have we dropped a test?

@michael-bryson We're not expecting to lose tests in the migration to JUnit 5.  I just wanted to point it out as it will be an important thing to verify when this ticket gets into QA.

**Comment by raghavendran-gopalakrishnan** [comment]
> The main branch build shows 42 tests but your build only shows 41:
> 
> * https://sou-jenkins2.swinfra.net/job/CAFapi/job/CAFapi~caf-common~develop~CI/88/testReport/
> * https://sou-jenkins2.swinfra.net/job/CAFapi/job/CAFapi~caf-common~914108~CI/2/testReport/
> 
> Why have we dropped a test?
> 
> @michael-bryson We're not expecting to lose tests in the migration to JUnit 5. I just wanted to point it out as it will be an important thing to verify when this ticket gets into QA.

Looks like surefire is using 

> Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider

instead of 
> Using auto detected provider org.apache.maven.surefire.testng.TestNGProvider

So the test was not picked up, I will look into this.

**Comment by raghavendran-gopalakrishnan** [comment]
> Looks like surefire is using
> 
> > Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
> 
> instead of
> 
> > Using auto detected provider org.apache.maven.surefire.testng.TestNGProvider
> 
> So the test was not picked up, I will look into this.

`junit-platform-commons` is included as transitive dependency by which `JUnitPlatformProvider` takes priority over `TestNGProvider`.

**Comment by raghavendran-gopalakrishnan** [comment]
Earlier dropped test is now picked up:
image

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
Just scrap the `util-testing` module and move the `MultiThreadTestUtil` class into the test code of `util-process-identifier`, since that is the only place that it is used.  (It may be surprising but scrapping this module is not a breaking change.)

File: util-process-identifier/src/test/java/com/hpe/caf/util/processidentifier/MultiThreadTestUtil.java [hunk 2]
```diff
@@ -13,9 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hpe.caf.util.testing;
+package com.hpe.caf.util.processidentifier;
```
Comment (dermot-hardy) [inline util-process-identifier/src/test/java/com/hpe/caf/util/processidentifier/MultiThreadTestUtil.java:16]: Does the class still need to be `public` in the new location?

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**